pmd-ui/src/main/resources/net/sourceforge/pmd/util/fxdesigner/fxml/auxclasspath-setup-popup.fxml

```diff
@@ -0,0 +1,66 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<?import javafx.geometry.Insets?>
+<?import javafx.scene.control.Button?>
+<?import javafx.scene.control.ListView?>
+<?import javafx.scene.control.SplitPane?>
+<?import javafx.scene.control.TitledPane?>
+<?import javafx.scene.layout.AnchorPane?>
+<?import javafx.scene.layout.VBox?>
+<VBox minWidth="-Infinity" snapToPixel="false" xmlns="http://javafx.com/javafx/9.0.1"
+      xmlns:fx="http://javafx.com/fxml/1" fx:controller="net.sourceforge.pmd.util.fxdesigner.AuxClassPathController">
+    <children>
+        <TitledPane collapsible="false" prefHeight="25.0" prefWidth="647.0" text="PMD Configuration"/>
+        <AnchorPane VBox.vgrow="ALWAYS">
+            <children>
+                <SplitPane dividerPositions="0.8407350689127105" minWidth="-Infinity" AnchorPane.bottomAnchor="-19.0"
+                           AnchorPane.leftAnchor="0.0" AnchorPane.rightAnchor="0.0" AnchorPane.topAnchor="0.0">
+                    <items>
+                        <VBox>
+                            <children>
+                                <ListView fx:id="fileListView" VBox.vgrow="ALWAYS"/>
+                            </children>
+                        </VBox>
+                        <VBox alignment="CENTER" SplitPane.resizableWithParent="false">
```

Comment:
The buttons on this UI are a bit overwhelming. Using icons would be better but I can do that change myself since it's not very interesting. What *you* should do is use content validation to enable the buttons only when they actually do something useful. The rest of the time they should be greyed out and inactive.

 You can use a ValidationSupport for that. There's an example for the property edition popup. See here we register some validators:

http://github.com/pmd/pmd/blob/master/pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/EditPropertyDialogController.java#L137-L157

And here we disable the commit button whenever preconditions are not fulfilled

http://github.com/pmd/pmd/blob/master/pmd-ui/src/main/java/net/sourceforge/pmd/util/fxdesigner/EditPropertyDialogController.java#L89

More documentation:

https://controlsfx.bitbucket.io/index.html?org/controlsfx/validation/ValidationSupport.html

Reply:
Also, there should be a cancel button available.